homeassistant/components/rest/__init__.py

```diff
@@ -54,12 +55,14 @@
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the rest platforms."""
-    component = EntityComponent[Entity](_LOGGER, DOMAIN, hass)
     _async_setup_shared_data(hass)
 
     async def reload_service_handler(service: ServiceCall) -> None:
         """Remove all user-defined groups and load new ones from config."""
-        if (conf := await component.async_prepare_reload()) is None:
+        conf = None
+        with contextlib.suppress(HomeAssistantError):
```

Comment:
Should we give them some type of feedback if there is something wrong with their yaml?

Reply:
The PR preserves the existing behavior which is asserted by tests. We might indeed want to let the error bubble up instead.

Reply:
LGTM, but it would be a nice improvement to tell them what's wrong.